vendor/engines/split_accounts/app/models/split_accounts/account_extension.rb

```diff
@@ -8,6 +8,27 @@ module AccountExtension
       has_many :parent_split_accounts, through: :parent_splits
 
       scope :excluding_split_accounts, -> { where("accounts.type != ?", "SplitAccounts::SplitAccount") }
+
+      after_save :update_suspended_at_for_parent_split_accounts
+      after_save :update_expires_at_for_parent_split_accounts
+    end
+
+    def update_suspended_at_for_parent_split_accounts
+      if parent_split_accounts.any? && suspended_at_changed?
+        parent_split_accounts.each do |split_account|
+          split_account.set_suspended_at_from_subaccounts
+          split_account.save
+        end
+      end
+    end
+
+    def update_expires_at_for_parent_split_accounts
+      if parent_split_accounts.any? && expires_at_changed?
+        parent_split_accounts.each do |split_account|
+          split_account.set_expires_at_from_subaccounts
```

Comment:
Does this end up running `set_expires_at_from_subaccounts` twice since it's also a `before_save` on the SplitAccount?